test/node/byte_utils.test.ts

```diff
@@ -0,0 +1,479 @@
+      expect(error).to.have.property('name', 'RangeError');
+    }
+  },
+  {
+    name: 'should throw if no argument is supplied',
+    // @ts-expect-error: testing bad input
+    inputs: [],
+    expectation({ error }) {
+      expect(error).to.have.property('name', 'TypeError');
+    }
+  },
+  {
+    name: 'should throw if invalid typed argument is supplied',
+    // @ts-expect-error: testing bad input
+    inputs: ['abc'],
+    expectation({ error }) {
+      expect(error).to.have.property('name', 'TypeError');
+    }
+  },
+  {
+    name: 'should handle zero sized allocation',
+    inputs: [0],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.have.property('byteLength', 0);
+    }
+  }
+];
+const equalsTests: ByteUtilTest<'equals'>[] = [
+  {
+    name: 'should return true for zero sized buffers',
+    inputs: [new Uint8Array(), new Uint8Array()],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.be.true;
+    }
+  },
+  {
+    name: 'should return true for equal buffers',
+    inputs: [Uint8Array.from([1, 2, 3]), Uint8Array.from([1, 2, 3])],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.be.true;
+    }
+  },
+  {
+    name: 'should return true for different Buffer instances whose contents are equal',
+    inputs: [Uint8Array.from([1, 2, 3]), Buffer.from([1, 2, 3])],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.be.true;
+    }
+  },
+  {
+    name: 'should return false for buffers with a different length',
+    inputs: [Uint8Array.from([1, 2]), Uint8Array.from([1, 2, 3])],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.be.false;
+    }
+  },
+  {
+    name: 'should return false for buffers with different content',
+    inputs: [Uint8Array.from([1, 2, 3]), Uint8Array.from([1, 2, 4])],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.be.false;
+    }
+  }
+];
+const fromNumberArrayTests: ByteUtilTest<'fromNumberArray'>[] = [
+  {
+    name: 'should construct a buffer from an array with numbers',
+    inputs: [[1, 2, 3]],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.deep.equal(Buffer.from([1, 2, 3]));
+    }
+  },
+  {
+    name: 'should construct an empty buffer from an empty array',
+    inputs: [[]],
+    expectation({ output, error }) {
+      expect(error).to.be.null;
+      expect(output).to.have.property('byteLength', 0);
+    }
+  }
+];
+const fromBase64Tests: ByteUtilTest<'fromBase64'>[] = [
+  {
+    name: 'should create buffer from base64 input',
+    inputs: [Buffer.from('abc\u{1f913}', 'utf8').toString('base64')],
+    expectation({ web, output, error }) {
+      if (!isNode16OrLater && web) {
+        // Skip reason: btoa and atob were not made globals until node 16
```

Comment:
for all the cases in this file that are skipped, is there a way to actually skip them or to alter the test case name to indicate what is actually tested; because here, for example, it seems that we actually won't be able to create the buffer from base 64 on Node 14 - is that true? or is it a test setup issue only?